litestar/middleware/exceptions/middleware.py

```diff
@@ -183,16 +187,31 @@ def __init__(self, app: ASGIApp, debug: bool | None, exception_handlers: Excepti
 
         .. deprecated:: 2.0.0
             The ``debug`` parameter is deprecated. It will be inferred from the request scope
+
+        .. deprecated:: 2.9.0
+            The ``exception_handlers`` parameter is deprecated. It will be inferred from the application or the
```

Comment:
If we only ever infer the exception handlers, this makes this middleware truly internal, as it's not really useful for an end user. Maybe we should deprecate this thing entirely then from the public interface?

Reply:
Is it currently public interface? TBH I added this b/c we'd done the same for the debug parameter, but the type isn't actually documented anywhere. Although I concede that not being in a private namespace muddies the water.

That aside, yeah, I agree we should deprecate the whole thing like CORS, b/c it requires special handling internally, and our use of it is hard coded. Unless there's a use case for adding extra EHM in the ASGI stack that I'm not seeing. The only thing I can think of is for catching errors from an interim middleware to covert it to a response before the next middleware. However, for that case I think it should be up to that middleware to convert expected errors into a response, and let unexpected errors propagate. Lo

Reply:
> However, for that case I think it should be up to that middleware to convert expected errors into a response, and let unexpected errors propagate.

Yeah, that sounds like the way to go

Reply:
There's a good example of this in our own code.

In using `SessionAuth`, we add `session_auth.on_app_init` to the `on_app_init` application parameter, which inserts `SessionAuth.middleware` as the first middleware on the app config object.

`SessionAuth.middleware` is actually an instance of `DefineMiddleware` that in turn instantiates an instance of `MiddlewareWrapper`.

On first call, `MiddlewareWrapper` builds a new middleware stack around the asgi app it has received on instantiation. It wraps `SessionAuthMiddleware` with `ExceptionHandlerMiddleware`, and then wraps that with 
`SessionMiddleware`.

Due to the fact that `SessionMiddleware` wraps `send()`, and we use raising `NotAuthorizedException` or `PermissionDeniedException` to communicate from `AbstractAuthenticationMiddleware.authenticate_request()` that a request has failed auth'n, we then need to _immediately_ wrap that middleware in the exception handler middleware so that the failed requests are turned into a response and we call `send()` _before_ the `SessionMiddleware` in the asgi stack has been bypassed.

So by only adding `SessionAuth` to an application, prior to this PR a connection would pass through:

`EHM-1 -> EHM-2 -> SM -> EHM-3 -> SAM -> EHM-4 -> Route.handle`

Where:

- EHM-1: outermost exception handler middleware, wrapping routing to capture 404 etc
- EHM-2: exception handler middleware that wraps the routed asgi middleware stack
- SM: session middleware
- EHM-3: exception handler middleware to catch auth exceptions and convert to a response/call `send()` before SM
- SAM: session auth middleware
- EHM-4: exception handler middleware to catch handler exceptions
- Route.handle: the inner most asgi app that calls the handler

If instead of raising `NotAuthorizedException`/`PermissionDeniedException`, `AbstractAuthenticationMiddleware.authenticate_request()` returned some other signal, e.g., ` -> AuthenticationResult | NotAuthorized | PermissionDenied`, and the middleware itself immediately constructed and awaited the response, there would be no need to wrap `SessionAuthMiddleware` in `ExceptionHandlerMiddleware`.

If that was the case, and after this PR, we'd have reduced the 4 layers of EHM, down to 2:

`EHM-1 -> SM -> SAM -> EHM-2 -> Route.handle`